etl/etl.py

```diff
@@ -14,15 +14,49 @@ def extract_from_nix():
     )
     print("Downloading data from nix as a dataframe...")    
     result = subprocess.run('NIXPKGS_FLAKE_REF="github:nixos/nixpkgs/master" nix eval --json --file '+ str(nixpkgs_graph_nix_file_path) + ' --extra-experimental-features "nix-command flakes"', shell=True, stdout=subprocess.PIPE)
-    dataframe = pd.read_json(result.stdout.decode(), orient="records")    
+    dataframe = pd.read_json(result.stdout.decode(), orient="records")
     return dataframe
 
 def data_process(dataframe):
-    dataframe["path"] = dataframe.groupby(["outputPath"])["path"].transform(lambda x : ", ".join(x))
+    dataframe["path"] = dataframe.groupby(["outputPath"])["path"] \
+                                 .transform(lambda x : ", ".join(x))
     dataframe = dataframe.loc[dataframe.astype(str).drop_duplicates().index]
     dataframe["path"] = dataframe["path"].fillna("")
+    dataframe = dataframe.reset_index(drop=True)
+    outputName = get_outputNames(dataframe)
+    dataframe["buildInputsName"] = \
+        dataframe["buildInputs"].apply((lambda x : \
+                                        path_to_name(x, outputName)))
+    dataframe["propagatedBuildInputsName"] = \
+        dataframe["propagatedBuildInputs"].apply((lambda x : \
+                                                  path_to_name(x, outputName)))
     return dataframe
 
+def get_outputNames(dataframe):
+    outputNameSet = set()
+    for name in dataframe["outputNameAll"].drop_duplicates():
+        outputNameSet = outputNameSet.union(set(name))
+    return outputNameSet
+
+def path_to_name(x: list, outputName):
```

Comment:
* snake_case
* type hints
```suggestion
def path_to_name(x: list, output_names: List[str]):
```